warehouse/models/mart/transit_database/dim_mobility_mart_providers.sql

```diff
@@ -0,0 +1,160 @@
+--
+-- We cannot use `_is_current` here because every year is marked as "current"
+-- since it's the "current" record for the respective year.
+annual_ntd AS (
+    SELECT * FROM {{ ref('dim_annual_ntd_agency_information') }}
+    WHERE state = "CA"
+
+    -- We only want data from the latest data from NTD. In the rare edge case
+    -- that an NTD ID is removed and no longer exists in the current year's
+    -- dataset, then replace `DENSE_RANK()` with `ROW_NUMBER()` and add a
+    -- `PARTITION BY ntd_id` inside of the `OVER()`.
+    QUALIFY DENSE_RANK() OVER(ORDER BY year DESC) = 1
+),
+
+gtfs_data_by_organization AS (
+    SELECT
+        orgs.key AS org_key,
+        ARRAY_AGG(gtfs_datasets.uri) AS gtfs_uris
+    FROM organizations orgs
+    LEFT JOIN
+        orgs_services_bridge ON orgs.key = orgs_services_bridge.organization_key
+    LEFT JOIN
+        gtfs_services ON orgs_services_bridge.service_key = gtfs_services.service_key
+    LEFT JOIN
+        gtfs_datasets ON gtfs_services.gtfs_dataset_key = gtfs_datasets.key
+    WHERE
+        orgs.public_currently_operating IS TRUE
+        AND gtfs_services.customer_facing IS TRUE
+        -- We only want to track schedule feeds for right now
+        AND gtfs_datasets.type = 'schedule'
+    GROUP BY
+        orgs.key
+),
+
+serviced_counties_by_organization AS (
+    SELECT
+        orgs.key AS org_key,
+        ARRAY_CONCAT_AGG(services.operating_counties) AS operating_counties
+    FROM organizations orgs
+    LEFT JOIN
+        orgs_services_bridge ON orgs_services_bridge.organization_key = orgs.key
+    LEFT JOIN
+        services ON services.key = orgs_services_bridge.service_key
+    GROUP BY
+        orgs.key
+),
+
+mobility_market_providers AS (
+    SELECT
+        annual_ntd.agency_name AS agency_name,
+        annual_ntd.ntd_id AS ntd_id,
+        annual_ntd.city AS hq_city,
+        orgs_hq_bridge.county_geography_name AS hq_county,
+        ARRAY(
+            SELECT DISTINCT
+                county
+            FROM UNNEST(serviced_counties_by_organization.operating_counties) county
+            ORDER BY county
+        ) AS counties_served,
+        annual_ntd.url AS agency_website,
+        county_geogs.caltrans_district AS caltrans_district_id,
+        county_geogs.caltrans_district_name AS caltrans_district_name,
+        orgs.is_public_entity AS is_public_entity,
+        orgs.public_currently_operating AS is_publicly_operating,
+        ARRAY(
+            SELECT DISTINCT
+            source
+            FROM UNNEST(funding_by_organization.funding_sources) source
+            ORDER BY source
+        ) AS funding_sources,
+        annual_ntd.voms_do AS on_demand_vehicles_at_max_service,
+        annual_ntd.total_voms AS vehicles_at_max_service,
+        ARRAY(
+            SELECT DISTINCT
+            uris
+            FROM UNNEST(gtfs_data_by_organization.gtfs_uris) uris
+            ORDER BY uris
+        ) AS gtfs_schedule_uris
```

Comment:
Ok this is a very weird quirk of our data model, but I think that `uri` may not be the best field here, depending on what you're aiming for; the field that Cal-ITP maintains for data scraping is `pipeline_url` but unfortunately that's not available in the mart -- you can recover it by decoding `base64_url` using our macro: `{{ from_url_safe_base64(base64_url) }}`. But I am not sure if the `uri` field is actually maintained at this point... I will write a ticket to consider removing it from the model because it is deceptive.